src/werkzeug/sansio/multipart.py

```diff
@@ -0,0 +1,251 @@
+import re
+from dataclasses import dataclass
+from enum import auto
+from enum import Enum
+from typing import cast
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from .._internal import _to_bytes
+from .._internal import _to_str
+from ..datastructures import Headers
+from ..exceptions import RequestEntityTooLarge
+from ..http import parse_options_header
+
+
+@dataclass
```

Comment:
Is the overhead (?) of creating `dataclass` instances needed for the events since they're relatively simple and only checked with `isinstance`? Especially since it requires a backport dependency.

Reply:
I think yes, `frozen` is useful (see new commit), as is the auto init, comparison methods, and reprs. Happy to change though.